### DISCUSSION TITLE
Whether to consider adopting Archetype-based design？

### DISCUSSION THREAD

**Discussion by armorhu** [skypjack/entt#855] [General]
After I researched the ecs framework of ue4 and unity, 
I found that they both adopt the design concept based on archtype.
According to my current understanding, the design concept based on archtype will indeed bring better cache hit rate. 
Because it ensures that the component pool is in the same 16k memory block, it will be easier to fit into the same L1cache。

Not sure what your opinion is, looking forward to your opinion

**Comment by skypjack** [comment]
> Not sure what your opinion is, looking forward to your opinion

Opinions are for those who want to sell their point of view. Choices are for those who evaluate and decide for themselves.
I've written (and will write more) about this in the _ECS Back and Forth_ series [here](https://skypjack.github.io/). I tried to explain pros and cons of the different models too.
EnTT offers groups today and I'm working on a new _tool_ which isn't available out there as far as I know. Time will tell us if it's good or not.

Some posts about my recent thoughts and discoveries are missing though. My fault, a pandemic and a war aren't helping though.
I'll try to fill the gap as soon as possible. For what is worth, they'll mainly cover two topics:
* The idea that centralized models (i.e. that of Unity but also the EnTT registry) have some (many?) weaknesses that are more evident in large teams/projects (that is why EnTT also allows a _registry-less_ model now)
* Multi-type storage, that is, a fancy name for the new model above, sort of halfway between groups (the limitations of which it overcomes) and archetypes (the issues of which it solves hopefully)

Finally, you can find a bunch of other libraries [here](https://github.com/skypjack/entt/wiki/Similar-projects) if you want to change and/or ask their authors for an opinion.
You should only use EnTT if you're sure. If you think it will be the cause of your perf issues, please, I'm the first to tell you to **stop**!
